### PR TITLE
fix(analytics-chart): better legend label truncation [MA-2048]

### PR DESCRIPTION
**Pull request by filipgutica**
https://konghq.atlassian.net/browse/MA-2048

# Summary

Only truncate the label when the legend grows such that it begins to wrap to multiple rows.

calculate the number of row in the grid layout, representing
the legend by comparing the "top" position of each item
incrementing the numberOfRows each time there is a difference.

Use the number of rows to determine if a label should be
truncated or not.

## Truncate when list grows and wraps to a new row.
https://www.loom.com/share/c338d6643f9d4f46baf49de6245e468a

## Truncate on window resize if legend is forced into a new row.
https://www.loom.com/share/bf2b6c0d7a03428ea290f3059fe6d01d

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: packages/analytics/analytics-chart/src/components/chart-plugins/ChartLegend.vue [hunk 1]
```diff
@@ -54,8 +57,19 @@ const props = defineProps({
...
+  if (legendContainerRef.value) {
+    const scrollHeight = legendContainerRef.value.scrollHeight
+    if (scrollHeight > parseInt(KUI_SPACE_150, 10)) {
+      shouldTruncate.value = true
+    }
+  }
+}
 
-const itemsRef = toRef(props, 'items')
+watch(() => props.items, checkForWrap, { immediate: true })
```
Comment (adorack) [inline packages/analytics/analytics-chart/src/components/chart-plugins/ChartLegend.vue:72]: I think this should have a `flush: 'post`` -- experimenting with the sandbox, it doesn't actually trigger wrapping the instant that a dataset starts to trigger scrolling.  At the time this `watch` runs, the DOM hasn't updated yet.

**Review by adorack — CHANGES_REQUESTED** [review]
It looks like there's something a bit off with the alignment of long labels?

image

File: packages/analytics/analytics-chart/src/components/chart-plugins/ChartLegend.vue (hunk 1 above)
Comment (filipgutica) [reply packages/analytics/analytics-chart/src/components/chart-plugins/ChartLegend.vue:72]: that sounds good

**Comment by filipgutica** [comment]
> It looks like there's something a bit off with the alignment of long labels?
> 
> image

I'm not sure I follow regarding the alignment.

image

**Review by mihai-peteu — APPROVED** [review]

**Review by mihai-peteu — APPROVED** [review]

**Pull request merged**